ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -494,3 +494,19 @@ def test_ipa_certs_check_ipacertnsstrust(self):
                 check["kw"]["key"] in DEFAULT_PKI_KRA_CERTS
             )
         tasks.uninstall_master(self.master)
+
+
+class TestIpaHealthCheckOnReplica(TestIpaHealthCheck):
+
+    def __init__(self):
+        super(TestIpaHealthCheckOnReplica, self).__init__()
+
+    def test_run_with_stopped_master(self):
+        """
+        Test output of healthcheck where master IPA services are stopped.
+        """
+        self.master.run_command(["ipactl stop"], raiseonerr=True)
+        self.replicas[0].run_command(["ipa-healthcheck",
+                                      "--failures-only",
+                                      "--output-type human"],
+                                     raiseonerr=True)
```

Comment:
I think the sequence should be:

ipa-healthcheck --failures-only
ipactl stop
ipa-healthcheck --failures-only

This will catch false positives to show that the health is bad before even starting the test and it isn't a false positive.

Or if you want to get fancier you can run:

ipa-healthcheck --failures-only --source ipahealthcheck.meta.services --output-type human | grep -c 'not running'

And then assert that the value is the number of expected services (9).

raiseonerr=True is not technically necessary since it's the default but it's not wrong to include it here since you are specifically expecting an exception to be raised, you just aren't catching that exception and ensuring it is what you expct.

Reply:
so, if I understand correctly, I could do:

```
self.master.run_command(["ipactl stop"])
out = self.master.run_command(["ipa-healthcheck", "--failures-only", "--source", "ipahealthcheck.meta.services",  "--output-type",  "human" , "|",  "grep", " -c", " 'not running'"])
assert out.stdout == "9"
```

is 9 constant? If not, is it possible to obtain it?

Reply:
Good point, it depends on what optional services are installed.

Since are are just ensuring that healthcheck will only execute the status commands and not throw a ton of false positives a regex to ensure that all lines are something like:

^ERROR: .*: not running$